src/WebApiContrib.Core/Binding/GlobalActionFilterApplicationModelConvention.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+using WebApiContrib.Core.Filters;
+using Microsoft.AspNetCore.Mvc.Internal;
+using System.Net.Http;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace WebApiContrib.Core.Binding
+{
+    public class GlobalActionFilterApplicationModelConvention : IApplicationModelConvention
+    {
+        private readonly ActionFilterAttribute _filter;
+        private readonly Func<ControllerModel, bool> _controllerPredicate;
+        private readonly Func<ActionModel, bool> _actionPredicate;
+        private readonly Func<SelectorModel, bool> _selectorPredicate;
+
+        public GlobalActionFilterApplicationModelConvention(
+            Func<ControllerModel, bool> controllerPredicate, Func<ActionModel, bool> actionPredicate, Func<SelectorModel, bool> selectorPredicate, ActionFilterAttribute filter)
```

Comment:
it should be more generic and you should use `IFilterMetadata` instead of `ActionFilterAttribute`. Same on the extension method. This will cover all possible filter types